Browser_IDE/editorMain.js

```diff
@@ -1,39 +1,31 @@
 "use strict";
-
+let executionEnviroment = new ExecutionEnvironment(document.getElementById("ExecutionEnvironment"));
+let storedProject = new IDBStoredProject(makeNewProject);
 // ------ Setup UI ------
 
-function setupCodeArea(element){
-    let editor = CodeMirror.fromTextArea(element, {
-        mode: "text/javascript",
-        theme: "dracula",
-        lineNumbers: true,
-        autoCloseBrackets: true,
-        styleActiveLine: true,
-        extraKeys: {"Ctrl-Space": "autocomplete"},
-        hintOptions: {
-            alignWithWord: false,
-            completeSingle: false,
-            useGlobalScope: false,
-        },
-        foldGutter: true,
-        gutters: ["CodeMirror-linenumbers", "CodeMirror-foldgutter"]
-    });
-
-    editor.on('inputRead', (cm, change) => {
-        if (!cm.state.completeActive) {
-            cm.showHint();
-        }
-    });
-    return editor;
+let editors = []
+let editorsMap={};
+let editorPathMap = {};
+function addNewEditor(filename, codeblock_name,eleID,ExecutionEnvironment, StoredProject,filePath  ) {
+    let elementID = document.getElementById(eleID);
```

Comment:
This element needs to be _created_ dynamically, not merely retrieved.

Consider: With your code, if someone wished to add a new editor tab, they would need to not only use this `addNewEditor` function, but also manually add a new element to `index.html`. The point is that `addNewEditor` should do this for us.